core/src/main/java/com/sequenceiq/cloudbreak/service/freeipa/FreeIpaConfigProvider.java

```diff
@@ -44,13 +45,23 @@
     private Optional<InstanceMetaDataResponse> selectInstance(DescribeFreeIpaResponse freeIpaResponse) {
         List<InstanceMetaDataResponse> metaDataResponses = freeIpaResponse.getInstanceGroups().stream()
                 .flatMap(ig -> ig.getMetaData().stream()).collect(Collectors.toList());
+        Optional<InstanceMetaDataResponse> pgwInstance = selectPgwInstance(metaDataResponses);
         Optional<InstanceMetaDataResponse> instanceMetaDataResponse =
-                chooseFirstOrderedByFqdnAndByStateIfPresent(metaDataResponses, Optional.of(InstanceStatus.CREATED))
-                        .or(() -> chooseFirstOrderedByFqdnAndByStateIfPresent(metaDataResponses, Optional.empty()));
+                pgwInstance.or(() -> chooseFirstOrderedByFqdnAndByStateIfPresent(metaDataResponses, Optional.of(InstanceStatus.CREATED))
+                        .or(() -> chooseFirstOrderedByFqdnAndByStateIfPresent(metaDataResponses, Optional.empty())));
```

Comment:
In case the primary gateway cannot be selected, will the selected node's InstanceMetaDataType be changed to GATEWAY_PRIMARY (is it needed to be adjusted at all?)?

Reply:
why would CB adjust the metadatatype of a FreeIPA instance? if PGW instance is down, it will be replaced/repaired in FMS at some point